extensions/wikia/ARecoveryEngine/ResourceLoaders/PageFair/ResourceLoaderAdEnginePageFairDetectionModule.php

```diff
@@ -0,0 +1,50 @@
+<?php
```

Comment:
this class looks very similar to `ResourceLoaderAdEngineSourcePointBase` and `ResourceLoaderAdEngineSourcePointDetectionModule`. How about designing the code in a different way to avoid duplications? However, it's just a suggestion I won't be stubborn about it.